Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -0,0 +1,140 @@
+import MongoSwiftSync
+import TestsCommon
+
+struct UnifiedTestRunner {
+    let internalClient: MongoClient
+    let serverVersion: ServerVersion
+    let topologyType: TestTopologyConfiguration
+
+    static let minSchemaVersion = SchemaVersion(rawValue: "1.0.0")!
+    static let maxSchemaVersion = SchemaVersion(rawValue: "1.0.0")!
+
+    init() throws {
+        let connStr = MongoSwiftTestCase.getConnectionString(singleMongos: false).toString()
+        self.internalClient = try MongoClient.makeTestClient(connStr)
+        self.serverVersion = try self.internalClient.serverVersion()
+        self.topologyType = try self.internalClient.topologyType()
+
+        // The test runner SHOULD terminate any open transactions using the internal MongoClient before executing any
+        // tests. Using the internal MongoClient, execute the killAllSessions command on either the primary or, if
+        // connected to a sharded cluster, all mongos servers.
+        switch self.topologyType {
+        case .single:
+            return
+        case .replicaSet:
+            let admin = self.internalClient.db("admin")
+            for address in MongoSwiftTestCase.getHosts() {
+                let isMaster = try admin.runCommand(["isMaster": 1], on: address)["ismaster"]!.boolValue!
+                if isMaster {
```

Comment:
instead of iterating over the hosts like this, could we just use a primary read preference?